lib/new_relic/telemetry/ecto/metadata.ex

```diff
@@ -1,16 +1,49 @@
 defmodule NewRelic.Telemetry.Ecto.Metadata do
   @moduledoc false
 
-  @postgrex_select ~r/FROM "(?<table>\w+)"/
-  @postgrex_insert ~r/INSERT INTO "(?<table>\w+)"/
-  @postgrex_update ~r/UPDATE "(?<table>\w+)"/
-  @postgrex_delete ~r/FROM "(?<table>\w+)"/
-  @postgrex_create_table ~r/CREATE TABLE( IF NOT EXISTS)? "(?<table>\w+)"/
   def parse(%{
         query: query,
         result: {_ok_or_error, %{__struct__: struct}}
       })
       when struct in [Postgrex.Result, Postgrex.Error] do
+    parse_query(query, "Postgres")
+  end
+
+  def parse(%{
+        query: query,
+        result: {_ok_or_error, %{__struct__: struct}}
+      })
+      when struct in [MyXQL.Result, MyXQL.Error] do
+    parse_query(query, "MySQL")
+  end
+
+  def parse(%{
+        query: query,
+        result: {_ok_or_error, %{__struct__: struct}}
+      })
+      when struct in [Exqlite.Result, Exqlite.Error] do
+    parse_query(query, "SQLite3")
+  end
+
+  def parse(%{query: query, repo: repo, result: {_ok_or_error, _}}) do
+    case repo.__adapter__() do
+      Ecto.Adapters.Jamdb.Oracle -> parse_query(query, "Oracle")
+      _ -> :ignore
+    end
+  end
+
+  def parse(%{result: {:ok, %{__struct__: Postgrex.Cursor}}}), do: :ignore
+  def parse(%{result: {:ok, %{__struct__: MyXQL.Cursor}}}), do: :ignore
+  def parse(%{result: {:ok, %{__struct__: SQLite3.Cursor}}}), do: :ignore
+  def parse(%{result: {:ok, nil}}), do: :ignore
```

Comment:
Although not included in this PR, a matching error may still occur in this part. Can I modify it like this? 
```suggestion
  def parse(%{result: {:ok, _}}), do: :ignore
```